### ISSUE TITLE
Null ref: T2.AlterantExtension_KatanaTeleport.OnCastOrSwing

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#555]
Reported in 0.1.3c.

Feedback ID: 20200529-5BDG9

```
5:04:27, Frame 348374, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.AlterantExtension_KatanaTeleport.OnCastOrSwing (T2.Creature targetCreature, T2.SeedFactory seedFactory)
T2.AlterantTracker.OnCastOrSwing (T2.Creature targetCreature, T2.SeedFactory seedFactory)
T2.MeleeWeapon.OnAttacked ()
T2.Weapon.OnUpdate ()
T2.CreatureBody.Update ()

GameStates: RompingThroughDungeon, GameOverScreen
GameSetupMode: Singleplayer
Players: 1
DungeonRoomTransitionerState: Null
Room: 7 / 7
RoomModifiers: MediumRoom (中型房间)
YouHaveDiedState: BecomingVisible
```

**Comment by ChrisNZL** [comment]
v0.1.4a

Feedback ID: 20200617-M6DQ9

```
12:50:10, Frame 304036, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.AlterantExtension_KatanaTeleport.OnCastOrSwing (T2.Creature targetCreature, T2.SeedFactory seedFactory)
T2.AlterantTracker.OnCastOrSwing (T2.Creature targetCreature, T2.SeedFactory seedFactory)
T2.MeleeWeapon.OnAttacked ()
T2.Weapon.OnUpdate ()
T2.CreatureBody.Update ()

GameStates: RompingThroughDungeon, GameOverScreen
GameSetupMode: Singleplayer
Players: 1
DungeonRoomTransitionerState: Null
Room: 6 / 6
RoomModifiers: LargeRoom (大型房间)
YouHaveDiedState: InitialDelay
```

**Comment by ChrisNZL** [comment]
Should be fixed in 0.1.4c.

Added many null checks.
